### PR TITLE
General fixes and sanity checks to improve !mp commands

### PR DESCRIPTION
**Pull request by lenforiee**
Referring to #110, it turns out that not only `!mp start` command was broken, but the rest of the commands were also either bloated, unusable or broken in some ways.

This PR is a collection of fixes to the commands while also making them usable for normal players now (bancho also allows for that).
I also decided to rename `#multi_` channel names to `#mp_` to allow for bancho-like IRC experience

### CODE REVIEW

**Comment by lenforiee** [comment]
The code has been tested locally, and everything seems to work

File: constants/chatbotCommands.py [hunk 1]
```diff
@@ -1761,98 +1791,105 @@ async def mpMake() -> Optional[str]:
...
-    #             f"channel from IRC, use /join #multi_{matchID} instead.",
-    #         )
-
-    #     await osuToken.joinMatch(userToken["token_id"], matchID)
-    #     return f"Attempting to join match #{matchID}!"
-
-    async def mpClose() -> Optional[str]:
-        if not (userID := await userUtils.getIDSafe(fro)):
-            raise exceptions.userNotFoundException("No such user")
+        async with redisLock(f"{match.make_key(multiplayer_match['match_id'])}:lock"):
```
Comment (tsunyoku) [inline constants/chatbotCommands.py]: i'm not necessarily challenging your choice, i just want to make sure we're aligned - what did you add the lock for here?

File: constants/chatbotCommands.py [hunk 2]
```diff
@@ -2354,66 +2455,85 @@ async def mpMods() -> Optional[str]:
...
-        if (colour := message[2].lower().strip()) not in {"red", "blue"}:
+        colour = message[2].lower().strip()
+        if colour not in {"red", "blue"}:
             raise exceptions.invalidArgumentsException(
                 "Team colour must be red or blue.",
             )
 
-        if not (userID := await userUtils.getIDSafe(username)):
-            raise exceptions.userNotFoundException("No such user.")
+        colour_const = {"red": matchTeams.RED, "blue": matchTeams.BLUE}[colour]
```
Comment (tsunyoku) [inline constants/chatbotCommands.py]: maybe simplify to `dict.get(colour)` and then raising if None

File: objects/match.py [hunk 3]
```diff
@@ -873,6 +875,11 @@ async def userLeft(match_id: int, token_id: str, disposeMatch: bool = True) -> N
         user_id=-1,
     )
 
+    await osuToken.update_token(
```
Comment (tsunyoku) [inline objects/match.py:882]: was this really missing? :sob:

**Review by tsunyoku — COMMENTED** [review]
looks good overall, might be nice to move `f"{match.make_key(multiplayer_match['match_id'])}:lock"` into a function though

File: objects/match.py (hunk 3 above)
Comment (lenforiee) [reply objects/match.py:882]: yep, probably more things are missing in the whole code base but this caused an issue in my testing

File: constants/chatbotCommands.py (hunk 1 above)
Comment (lenforiee) [reply constants/chatbotCommands.py]: good catch, i don't think it's needed there i think i just rush added it

Comment (lenforiee) [reply constants/chatbotCommands.py]: resolved in 2e0740d

File: constants/chatbotCommands.py (hunk 2 above)
Comment (lenforiee) [reply constants/chatbotCommands.py]: resolved in 38649d3

**Comment by lenforiee** [comment]
Moved the redis lock key creation to its own function in e92d27a, we should also do that for user token some day

**Review by tsunyoku — APPROVED** [review]
well lets see how this does

**Pull request merged**